src/main/java/org/apache/sling/feature/launcher/impl/Main.java

```diff
@@ -312,8 +359,15 @@ public static void main(final String[] args) {
 
         // check if launcher has already been created
         final LauncherConfig launcherConfig = new LauncherConfig();
-        parseArgs(launcherConfig, args);
-
+        Options options=  parseArgs(launcherConfig, args);
+
+        if(launcherConfig.getFeatureFiles().isEmpty()) {
+            Main.LOG().warn("Missing Feature-File(s). Please set an Feature-File to run.");
+            
```

Comment:
I think this is wrong - we support starting up with no features provided it is a restart. In that case, we load the previously assembled feature: https://github.com/apache/sling-org-apache-sling-feature-launcher/blob/3cefbd39ed17a9c880a07e2b4d7d5a1ce1526e35/src/main/java/org/apache/sling/feature/launcher/impl/Bootstrap.java#L195

I think if you want todo this, it would need to do the bootstrap.run() and catch the exception from here: https://github.com/apache/sling-org-apache-sling-feature-launcher/blob/3cefbd39ed17a9c880a07e2b4d7d5a1ce1526e35/src/main/java/org/apache/sling/feature/launcher/impl/Bootstrap.java#L201

If that exception happens, you could do the printHelp...

Reply:
I would like to handle this in the Main. There it is easier and I catched the IllegalStateException.
Hope this is also okay for you

Reply:
thats is fine - maybe re-throw the exception if it isn't because of a no features (just in case it had another cause).

Reply:
my last commit wasn't correct, because the Exception was catched before.
I think it would be better this way.